### PR TITLE
feat: extra APIs isVue2 and isVue3, iife build improves

### LINKED ISSUE #11: check version of Vue
Hi.
At first, great library and idea!

I have a feature request.
It would be nice to be able to check the version of the Vue because there are still some differences.

Something like this maybe.

```js
import {h, isVue2} from 'vue-demi';

h('div', {
  style: {/*...*/},
  ...(isVue2 ? {on: {click: onClick}} : {onClick}),
})
```

### PR DESCRIPTION
**Pull request by SeregPie**
I tried to create a rollup config to create all the files automatically but it seems impossible. To create `.d.ts` files I need to have both Vue versions at the same time. For `iife` I need some sort of dynamic require which is not well done with rollup.

---

I unified all exports in all files, so Vue is always exported. So you could do that:

```js
import {Vue, isVue2} from 'vue-demi'

if (isVue2) {
  Vue.set(state, 'n', 1)
} else {
  state.n = 1
}
```

---

There is no need to check if VueCompositionAPI is already installed, because:
"When this method is called on the same plugin multiple times, the plugin will be installed only once."
https://vuejs.org/v2/api/#Vue-use

---

Also closes #11

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,9 +3,9 @@
   "version": "0.3.3",
   "main": "lib/index.cjs.js",
   "typings": "lib/index.d.ts",
-  "module": "lib/index.esm.js",
-  "unpkg": "lib/index.iffe.js",
-  "jsdelivr": "lib/index.iffe.js",
+  "module": "lib/index.es.js",
+  "unpkg": "lib/index.iife.js",
+  "jsdelivr": "lib/index.iife.js",
```
Comment (antfu) [inline package.json:8]: Thanks for spotting out the `iife` typo, but I would like to keep the `esm` naming.

File: lib/v2/index.es.js [hunk 2]
```diff
@@ -1,9 +1,12 @@
 import Vue from 'vue'
 import VueCompositionAPI from '@vue/composition-api'
 
-if (Vue && !Vue['__composition_api_installed__'])
-  Vue.use(VueCompositionAPI)
+Vue.use(VueCompositionAPI)
```
Comment (antfu) [inline lib/v2/index.es.js:4]: Can you revert this? It's necessary.

File: lib/index.iife.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+(function(window) {
+  var VueDemi = {};
+  var Vue = window.Vue;
+  if (Vue) {
+    if (Vue.version.startsWith('2.')) {
```
Comment (antfu) [inline lib/index.iife.js:5]: There is no `startsWith` in IE11, should check for `[0] === '2'` and `[1] === '.'`

File: lib/index.iife.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+      var VueCompositionAPI = window.VueCompositionAPI;
+      if (VueCompositionAPI) {
+        Vue.use(VueCompositionAPI);
```
Comment (antfu) [inline lib/index.iife.js:8]:
```suggestion
```

No need, VCA will register itself in cdn mode.

File: lib/v2/index.cjs.js [hunk 5]
```diff
@@ -1,14 +1,10 @@
 const Vue = require('vue')
 const CompositionAPI = require('@vue/composition-api')
 
-if (Vue && !Vue['__composition_api_installed__'])
-  Vue.use(CompositionAPI)
+Vue.use(CompositionAPI)
```
Comment (antfu) [inline lib/v2/index.cjs.js:4]: Revert this as well please.

File: lib/v2/index.d.ts [hunk 6]
```diff
@@ -1,4 +1,7 @@
 import Vue from 'vue'
+declare const isVue2 = true
```
Comment (antfu) [inline lib/v2/index.d.ts:2]:
```suggestion
declare const isVue2: boolean = true
```

Same as `v2`

**Review by antfu — CHANGES_REQUESTED** [review]
Feels like since we have `isVue2`, we should have the `isVue3` as well.

File: README.md [hunk 7]
```diff
@@ -47,6 +47,43 @@ import { ref, reactive, defineComponent } from 'vue-demi'
...
+          : {
+            onClick,
+            ref: buttonRef,
+          }
+        ),
+      })
+    })
+  }
+}
+```
```
Comment (antfu) [inline README.md:84]:
```suggestion
### Extra APIs

Vue Demi provides extra APIs `isVue2` `isVue3` to help distinguishing user versions.

```ts
import { isVue2 } from 'vue-demi'

if (isVue2) {
  // vue 2 only logic
} else {
  // vue 3 only logic
}
```

**Comment by SeregPie** [comment]
Ok. Thanks for the review.
I resolved everything so far. Also needed to rewrite `Object.assign` because it is also not supported in IE11.
Added `isVue3`.

File: lib/v2/index.cjs.js [hunk 8]
```diff
@@ -1,14 +1,12 @@
...
-  Vue.use(CompositionAPI)
+  Vue.use(VueCompositionAPI)
 
-Object.keys(CompositionAPI).forEach(key => {
-  // VCA exports it's own version, should not populate `Vue.version`
-  if (!Vue[key] && key !== 'version') { 
-    Vue[key] = CompositionAPI[key]
-  }
+Object.keys(VueCompositionAPI).forEach(key => {
+  exports[key] = VueCompositionAPI[key]
```
Comment (antfu) [inline lib/v2/index.cjs.js:8]: Should exclude VCA's `version` and export `exports.version = Vue.version`

File: lib/index.iife.js [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+      if (VueCompositionAPI) {
+        for (var key in VueCompositionAPI) {
+          VueDemi[key] = VueCompositionAPI[key];
+        }
+        VueDemi.isVue2 = true;
+        VueDemi.isVue3 = false;
+        VueDemi.Vue = Vue;
+      }
+    } else
+    if (Vue.version.slice(0, 2) === '3.') {
```
Comment (antfu) [inline lib/index.iife.js:16]:
```suggestion
    } else if (Vue.version.slice(0, 2) === '3.') {
```

File: lib/index.iife.js [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+      for (var key in Vue) {
+        VueDemi[key] = Vue[key];
+      }
+      VueDemi.isVue2 = false;
+      VueDemi.isVue3 = true;
+      VueDemi.Vue = Vue;
+    }
```
Comment (antfu) [inline lib/index.iife.js:23]:
```suggestion
    } else {
      console.error('[vue-demi] Vue version ' + Vue.version + ' unsupported.')
    }
```

File: lib/index.iife.js [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+  }
```
Comment (antfu) [inline lib/index.iife.js:24]:
```suggestion
  } else {
    console.error('[vue-demi] no Vue instance found, please be sure to import vue-demi after Vue.')
  }
```

File: lib/index.iife.js [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+        VueDemi.Vue = Vue;
+      }
+    } else
+    if (Vue.version.slice(0, 2) === '3.') {
+      for (var key in Vue) {
+        VueDemi[key] = Vue[key];
+      }
+      VueDemi.isVue2 = false;
+      VueDemi.isVue3 = true;
+      VueDemi.Vue = Vue;
```
Comment (antfu) [inline lib/index.iife.js:22]:
```suggestion
      VueDemi.Vue = Vue;
      VueDemi.version = Vue.version;
```

File: lib/index.iife.js [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+  if (Vue) {
+    if (Vue.version.slice(0, 2) === '2.') {
+      var VueCompositionAPI = window.VueCompositionAPI;
+      if (VueCompositionAPI) {
+        for (var key in VueCompositionAPI) {
+          VueDemi[key] = VueCompositionAPI[key];
+        }
+        VueDemi.isVue2 = true;
+        VueDemi.isVue3 = false;
+        VueDemi.Vue = Vue;
```
Comment (antfu) [inline lib/index.iife.js:13]:
```suggestion
        VueDemi.Vue = Vue;
        VueDemi.version = Vue.version
```

File: README.md [hunk 14]
```diff
@@ -47,6 +47,20 @@ import { ref, reactive, defineComponent } from 'vue-demi'
...
+
+```ts
+import { isVue2 } from 'vue-demi'
+
+if (isVue2) {
+  // vue 2 only logics
+} else {
+  // vue 3 only logics
+}
+
```
Comment (antfu) [inline README.md:62]: Can you help add the ending backquote? Not able to do that by review suggestions 😅

File: README.md [hunk 15]
```diff
@@ -47,6 +47,20 @@ import { ref, reactive, defineComponent } from 'vue-demi'
 
 Publish your plugin and all is done!
 
+### Extra APIs
+
+Vue Demi provides extra APIs `isVue2` `isVue3` to help distinguishing user versions.
```
Comment (antfu) [inline README.md:52]:
```suggestion
Vue Demi provides extra APIs `isVue2` and `isVue3` to help distinguishing user versions.
```

**Review by antfu — CHANGES_REQUESTED** [review]

**Comment by antfu** [comment]
Thanks a lot!

**Comment by SeregPie** [comment]
Thank you too.

**Pull request merged**